From 9ebf7f07103677543d886f0f4aae3edc76bcc5b2 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Fri, 25 Oct 2013 13:12:28 +0800
Subject: [PATCH 036/115] ENGR00284503 Pop up "Cactus player has stopped." if
 close subtitle

subtitle format is unsupported. selectTrack fails but GMPlayer doesn't
return failure and apk is not aware of it.
When closing subtitle, apk calls deselectTrack and GMPlayer
finds that current subtitle number is -1 then return fail. This leads
to exception in apk.

Add exception handler for selectTrack&deselectTrack in CactusPlayer.

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 .../com/freescale/cactusplayer/VideoPlayer.java    | 31 +++++++++++-----------
 1 file changed, 16 insertions(+), 15 deletions(-)

diff --git a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
index 2695cf5..c6f25aa 100755
--- a/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
+++ b/CactusPlayer/src/com/freescale/cactusplayer/VideoPlayer.java
@@ -949,7 +949,13 @@ public class VideoPlayer extends Activity implements SeekBar.OnSeekBarChangeList
         else if(itemId == CLOSE_SUBTITLE)
         {
             if(mCurSubtitleTrack >= 0 && mMediaPlayer != null){
-                mMediaPlayer.deselectTrack(mCurSubtitleTrack);
+                Log.v(TAG,"to deselected track  " + mCurSubtitleTrack);
+                try{
+                    mMediaPlayer.deselectTrack(mCurSubtitleTrack);
+                }
+                catch(Exception e){
+                    Log.d(TAG, "Failed to deselect track !!!");
+                }
                 mCurSubtitleTrack = -1;
                 mCurSubtitleIndex = -1;
                 mSubtitleTextView.setText(null,0);
@@ -1264,24 +1270,19 @@ public class VideoPlayer extends Activity implements SeekBar.OnSeekBarChangeList
                     MediaPlayer.TrackInfo[] ti = mMediaPlayer.getTrackInfo();
                     int totalCount = ti.length;
                     for(int j = 0; j< totalCount ; j++){
+                        Log.v(TAG,"track " + j + " type is " + ti[j].getTrackType());
                         if(ti[j].getTrackType() == MediaPlayer.TrackInfo.MEDIA_TRACK_TYPE_TIMEDTEXT)
                             index++;
                         if(index == i){
-                            mMediaPlayer.selectTrack(j);
-                            /*
-                            Parcel request = Parcel.obtain();
-                            Parcel reply = Parcel.obtain();
-                            try {
-                                request.writeInterfaceToken(IMEDIA_PLAYER);
-                                request.writeInt(INVOKE_ID_SELECT_TRACK);
-                                request.writeInt(j);
-                                mMediaPlayer.invoke(request, reply);
-                            } finally {
-                                request.recycle();
-                                reply.recycle();
+                            Log.v(TAG,"selected track " + j);
+                            try{
+                                mMediaPlayer.selectTrack(j);
+                            }
+                            catch(Exception e)
+                            {
+                                Log.d(TAG, "Failed to select subtitle track!!!");
+                                break;
                             }
-                            */
-                            //mInfoView.setText();
                             mCurSubtitleTrack = j;
                             mCurSubtitleIndex = i;
                             break;
-- 
1.8.0

